processing/src/main/java/io/druid/segment/filter/Filters.java

```diff
@@ -100,19 +101,8 @@ public static Filter convertDimensionFilters(DimFilter dimFilter)
       filter = new SpatialFilter(spatialDimFilter.getDimension(), spatialDimFilter.getBound());
     } else if (dimFilter instanceof InDimFilter) {
       final InDimFilter inDimFilter = (InDimFilter) dimFilter;
-      final List<Filter> listFilters = Lists.transform(
-          inDimFilter.getValues(), new Function<String, Filter>()
-          {
-            @Nullable
-            @Override
-            public Filter apply(@Nullable String input)
-            {
-              return new SelectorFilter(inDimFilter.getDimension(), input);
-            }
-          }
-      );
 
-      filter = new OrFilter(listFilters);
+      filter = new InFilter(inDimFilter.getDimension(), Sets.newHashSet(inDimFilter.getValues()));
```

Comment:
`ImmutableSet.copyOf` should offer better performance than HashSet for this case, especially for single element sets